### ISSUE TITLE
Interactions with the list of the results

### ISSUE THREAD

**Issue by lilive** [lilive/Freeplane-Jumper#15]
Due to this [feature request](https://sourceforge.net/p/freeplane/discussion/758437/thread/8ea365816c/#238d/a26a) and this [discussion](https://github.com/lilive/Freeplane-Jumper/issues/13), I'm considering to change the interactions with the results list. Can you tell me what do you think about this proposal ?

Goals:
- New possibility to select many items in the result list. So Jumper may help to select nodes in a map.
- Allow the user to click an item to show it in the map without closing Jumper.

How to achieve these goals:

Keyboard interaction with the result list
         
- Up/Down arrow focus a result, and center the map view around it.
- Ctrl/Cmd + Space select/unselect a result
- Shift + Up/Down selected the focused result, focus the previous/next result, select it, and center the map view around it.
- Enter jump to the focused result. If there is some selected results, they are all selected in the map.
  
Mouse interaction with the result list
      
- Click a result focus it, and center the map view around it.
- Ctrl/Cmd + Click select/unselect a result.
- Shift + Click select all the results from the most recently selected to this one.
- Double-click a result jump to it. If there is some selected results, they are all selected in the map.

Thank you

**Comment by lilive** [comment]
> -Mnemonics:
> a loosely related suggestion, that could help with the selection work: adding mnemonics to the results list. I mean: if the user presses Alt, a list of numbers or letters appears besides each item of the list; so he can then press the corresponding number to hit the result. I miss it, because, sometimes, my desired result is in the middle of the list, so I need to go all over the results (or use the mouse, which is slow, too). Having 0 to 9 and then "a" to "z" would cover all the results.

Good idea. But Alt 1-2-3-4-5-6-7 are already assigned to the "Where to search" options. Perhaps with Ctrl ?

**Comment by lilive** [comment]
> -Persistent Search:
> a more advanced feature, but that would make Jumper the ultimate node selection tool: another way to select the results (besides the usual way), so they keep selected even when the search terms change.
> For example: the user searches for "Linux", then he uses this "persistent selection" on some results; then, without leaving Jumper, he searches for "Windows", then chooses more results; in the end, he will have a list with all the results selected in that Jumper session.

Do you think you will use this feature frequently ?

**Comment by lilive** [comment]
> I won't be able to comment for the next couple of weeks

Thank you for telling me.

**Comment by bepolymathe** [comment]
Hi @lilive, Hi @euu2021 

This discussion is exciting. For my part, I agree with the proposals mentioned by @euu2021 and especially with the persistent search because it would have the advantage of being faster and more flexible in the application of filters when different terms are often searched. 

Also, the idea of being able to make connections directly from Jumper would be wonderful and would usefully complement to Mapinsight. But there are several types of links (connectors, links by ID, clone, etc) maybe it's a lot of work 🤔 

The other interaction with the result list that I would find useful would be to allow copying the results and in particular the IDs (or even path+ID) to make cross-references in other software.

**Comment by ldmpub** [comment]
To help you define dev priorities, according to me +1 on: 
* **Focus a result can center the map view around it** (the node), 2 ways to perform this
  * Keyboard: use Up/Down arrow to navigate between Jumper result lines to focus
  * Mouse: user can click on a Jumper result line to focus.

Pressing enter when a line is focused (Keyboard trigger) or pressing Ctrl/Cmd + mouse click (Mouse trigger)=> jumps in the map with node centered closing Jumper Dialog.

**Comment by lilive** [comment]
Hello,
This quick message to give you some news : I have worked on Jumper from time to time, in rare occasions, and the selection of multiple results is almost ready, but not enough to be released.
